pcap.go

```diff
@@ -9,17 +9,12 @@ package pcap
 #include <ifaddrs.h>
 
 #ifdef  __GLIBC__
-// See this for glibc 2.14 hack below
-// https://www.win.tue.nl/~aeb/linux/misc/gcc-semibug.html
-
-void *__memcpy_glibc_2_2_5(void *, const void *, size_t);
-
 // We use memmove rather than memcpy here since it is safer
 // for overlapping memory regions.
-asm(".symver __memcpy_glibc_2_2_5, memmove@GLIBC_2.2.5");
+
```

Comment:
can pcap.go be made arch specific?

Reply:
Duplicating the whole file for memmove seems not ideal.
I could not find a reason why memmove@2.2.5 was being used. I read the link in the code and it points to a problem in memcpy not memmove. Even memcpy problem would happen in machines who have glibc version < 2.14 which was released 9 yrs ago.
Is there a reason we should preserve this versioned symbol?

Reply:
glibc 2.2.5 was released in 2002 and we are pinning memmove to that for no apparent reason ? :)

Reply:
Correctness is more important here. The onus is to ensure memmove of "current" glibc is not going to lead us into data correctness problem both for the memmove and memcpy usage.

Reply:
why do you think current memmove will lead us into correctness issue vs 19 yr old one?

Reply:
Is there no change in memmove/memcpy since GLIBC 2.2.5?

Reply:
does it matter? Pinning to an older version is a bigger risk when we don't have a reason to.
I have tested connections with this change and it works fine.

Reply:
I have moved this conversation to the bug because it belongs there